### PR TITLE
Scatter laser shells now use the scatter laser beam, and makes them significantly easier to make. Projectiles can now have damage falloff.

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

Allows for damage falloff to apply to more than just shotgun pellets. Now any projectile can have a damage falloff defined.

Scatter Laser shells no longer use the minigun beams to determine their damage. Instead they use the actually defined scatter laser beams.  Those beams do 7.5 damage per pellet, times by 6 pellets.

Scatter laser beams now have damage falloff, a separately defined (positive) wounding power from normal beams, and wound falloff.

Scatter laser shells can be printed from security protolathes once you have weapon tech.

Scatter laser shells _may_ be damaged by EMPs based on severity. The result is that it fires a practically useless volley of laser fire. They cause a honk sound when they hit, so you know when you've shot one of these.

## Why It's Good For The Game

Well, we want shotguns universally to not be defined by their damage output (especially extreme damage output) but by niche.

What does the scatter laser shell currently occupy as a niche?

The single highest damage output of any projectile weapon in direct damage. The thing we don't want of shotguns, and it is reigning champion of all guns. 

Okay, that's a bit misleading, because obviously it is competing with the likes of .50 BMG which does 70 damage outright and dismembers limbs, potentially doing upwards of 90 damage if it does, and also hard stuns people. Obviously _that_ is technically a stronger bullet.

But not for raw damage, because the scatter laser does 90 damage out the gate, barring any potential wounding that might occur which increases the damage multiplicatively. No gimmicks, no extra procs, nothing. It's just 15 force lasers (with no damage dropoff) split between 6 beams.

And the reason for this is because this shell has been nerfed once prior by making it not fire 6 normal laser shots into someone. That was 120 damage at the time, 120 to 90 was...I guess a nerf during the taser era. Depends on how you viewed it. Buckshot was doing like 80 at the time, believe me it was a wild period. But anyway, when we did the whole damage rearrangement over the course of the laser few years, every other shell got touched except this one for some reason. Even pulse slugs lost 10 damage while this was still sitting on 90 force point blank.

So what is the new niche? Well, it's laser buckshot. That's not a niche but crew don't get buckshot, so this is their buckshot. It wounds real good. Real goddamn good. And its is a laser. It fits the aesthetic, obviously.

Okay, thanks.

## Changelog
:cl:
balance: Scatter laser shells actually utilize the _real_ scatter laser beam. This comes with damage changes. And wounding power.
feature: EMPs can potentially damage scatter laser shells.
refactor: All projectiles can now have damage falloff defined. Yay.
balance: Scatter laser shells can be printed when weapons technology is researched.
/:cl:

### CODE REVIEW

**Comment by YuliceTheDruid** [comment]
If I'm reading this correctly with my dumb no code knowing brain then this drops scatter lasers down to 30 burn if all pellets hit point blank? Less if you miss any or are tiles away?

**Comment by necromanceranne** [comment]
More if you wound but yes.

File: code/modules/projectiles/projectile/beams.dm [hunk 1]
```diff
@@ -94,6 +94,16 @@
 	name = "laser pellet"
 	icon_state = "scatterlaser"
 	damage = 5
+	wound_bonus = 5
+	bare_wound_bonus = 5
+	damage_falloff_tile = -0.45
+	wound_falloff_tile = -2.5
+
+/obj/projectile/beam/scatter/pathetic
```
Comment (CRITAWAKETS) [inline code/modules/projectiles/projectile/beams.dm:102]: should probably have a different icon and if possible a unique sound cue (just like a trumpet noise or fart sound)

Comment (necromanceranne) [reply code/modules/projectiles/projectile/beams.dm:102]: I was thinking the same thing.

**Comment by necromanceranne** [comment]
> I'm all in favor of this though I feel like scatter lasers should probably be more around raw damage than wounding, since that's their current niche (even if wound calcs make it currently wound more than this PR would make them do).

Raw damage just straight up can't be a defining aspect of shotgun shells. No chance. Literally any other feature should be considered first, and I'll be honest I had a bit of trouble trying to work out alternatives to scatter laser simply because it is so generically plain.

**Comment by Striders13** [comment]
If it's gonna have no damage point blank, you might as well don't give it damage falloff, so you can at least troll people with wounds from afar. Otherwise I don't see any use case for this.

**Comment by OverwatchVoice** [comment]
If shotgun shells are meant to fullfill unique, more niche applications, what exactly will these fill?
If you're going for wounds, hellfire lasers both deal more direct damage AND have a higher chance of inflicting catastrophic burns with more shots than a fully loaded shotgun. I guess it's nice to have a shotgun shell variant, at least?

**Comment by SimplySmiley123** [comment]
> > I'm all in favor of this though I feel like scatter lasers should probably be more around raw damage than wounding, since that's their current niche (even if wound calcs make it currently wound more than this PR would make them do).
> 
> Raw damage just straight up can't be a defining aspect of shotgun shells. No chance. Literally any other feature should be considered first, and I'll be honest I had a bit of trouble trying to work out alternatives to scatter laser simply because it is so generically plain.
> 
> Edit: to clarify what I'm basing this argument on, here is oranges [explanation](https://github.com/tgstation/tgstation/pull/78235#issuecomment-1712932668).

And without damage the shotguns just, don't get used at all.

The only time right now crew ever thinks of using a shotgun is when one seccie is stubborn enough to bug science hard enough to research tech shells and then make a dozen scatter laser shells or frag shells, and even then it's rare for those to get used anyway as by the time you get the research for those shells, the round might have _already ended._

**Comment by necromanceranne** [comment]
> If shotgun shells are meant to fullfill unique, more niche applications, what exactly will these fill? If you're going for wounds, hellfire lasers both deal more direct damage AND have a higher chance of inflicting catastrophic burns with more shots than a fully loaded shotgun. I guess it's nice to have a shotgun shell variant, at least?

Due to the way that pellet clouds work, it'll probably outpace hellfire for wound power due to calculating the final wound chance based on the number of pellets that hit the same limb. So, if we assume everything hits a single limb, the probability of a wound will usually outdo hellfire at point blank almost every time. And, because burn wounds are a multiplier for damage, usually it will kill quicker than the numbers suggest once you start mag dumping into someone.

Also due to how pellet clouds work, presently, against anyone in even a modicum of armor, scatter laser will almost always fail to wound. It usually just kills someone due to overwhelming damage against even targets with high armor values like elite suits.

**Comment by castawaynont** [comment]
these don't make them easier to make at all. nobody makes tech shells in the first place because of the exotic ammo requiring adv. weap. tech. by the time you can actually print tech shells tier 4 components have already been researched. this is basically just removing laser scatter, since there's no purpose to it now.

**Comment by Overuseddart** [comment]
By the time you’d convince sci to research tech shells, anyone you’d want to shoot this ammo at would have armor, making the gun do basically no damage because each pellet’s damage will be reduced. Also nobody is going to be standing point blank for you to shoot them with a shotgun. With much less effort you could simply buy a hellfire laser or even a cleaver for better damage.

**Comment by TheRyeGuyWhoWillNowDie** [comment]
You've outdone youself here. I know /tg/coders absolutely despise shotguns, but thirty damage at point blank. On a shotgun. With tech shells. Just remove shotguns already, they're so far past the "dead horse" phase that now they're a tube of elmer's glue, and you're still beating it.

**Comment by generalthrax** [comment]
While scatter laser's damage output is incredibly high, to get the absolute maximum you'd have to be right on top of them. This (can) be a bit more riskier than just unloading your pretty cracked laser into them, and I've definitely overextended before trying to get the point blank on targets only to get flanked.

Not denying that burn wounds _aren't_ dangerous, but having a specialized shell intended on dealing them feels a bit odd. They're fairly easily treated on the go, and compared to brute wounds caused by ballistics, do not completely fuck you over unless you're unloaded on immediately after (where you'd just die regardless of if you're wounded or not).

The power of wounding is very dependent on it's ability to hinder the person you're fighting and forcing them to get (often quite slow) treatment in fear of debilitating short-term consequences. Burn wounds are (comparatively) the least severe, and as stated above, are trivial to treat.

Have you tested these shells to see how comparable they are to other lethal options? What scenarios do you expect for sec to decide to use these over standard lasers? Officers frequently used these to trade risk for a higher potential damage, and I'm curious to see what role you intend these to fill now that damage has been considerably scaled down for higher wound chance on the weaker types of wounds.

**Comment by Jackal-boop** [comment]
I don't like this, scatter lasers are already rare enough and aren't over powered. As long as you keep a tile or 2 of distance you won't be zoned out instantly m

**Comment by necromanceranne** [comment]
So, some changes. Someone suggested just being able to print them, and I felt that was not a terrible concession. I do not like tech shells as they stand, it's a bit fiddly, and I was already looking to make these easier to make. So you can print the shells directly once you have the Weapons Technology node.

They do more damage. About 7.5*6 pellets = 45 damage overall. This value will fluctuate due to wounding being a force multiplier.

To clear up any confusion about how pellets work; when any given ammo casing fires multiple projectiles (like a shotgun does), it will automatically make that into a pellet cloud. It does not need to be the pellet subtype of projectile for this behaviour.

Pellet clouds add all the values of the individual projectiles together before applying their effects on a given limb. 

**Testing the shells by using past scatter laser against the proposed scatter laser, two things are observable;**

**Current scatter laser** cannot wound armored targets of about the average security officer level, even at point blank, but even the heftiest armor cannot withstand scatter laser fire most of the time. You are either immune to the damage, or you're dying to it very quickly. It'll also usually cause catastrophic wounds against a target with bare flesh, but that's nothing surprising given the damage inflicted and the lack of defenses, as well as the bare flesh bonus being as high as it is. Wounds at that point are largely meaningless given how swiftly it kills.

**For the proposed scatter laser change**, against armored targets of about sec level, it varies heavily. Sometimes it goes directly to catastrophic wounds on first shot, and followups result in about two more shots necessary to bring down the target. Sometimes it doesn't, and it takes about four shots (it does 33 damage on initial shot). Against unarmored targets, the catastrophic burn wound will be all but guaranteed, and so a followup shot will almost always put the target into crit if it is immediately followed up on. So, same crit time against unarmored targets _usually_. It will vary slightly due to wounds inherently being unpredictable.

Both of these are with respect of our potential target being near or literally shot pointblank. Once you're past point blank, every single shotgun pellet projectile varies considerabl;, either from bullet deviation or damage loss from travel time, to simply not having a tight enough cone to cause every projectile to land. But the proposal effectively results in a much swinger damage output if that makes sense.

**Comment by generalthrax** [comment]
This is based, and much better. Thanks for going through with testing

File: code/modules/projectiles/projectile/beams.dm (hunk 1 above)
Comment (necromanceranne) [reply code/modules/projectiles/projectile/beams.dm:102]: Resolved this for now

**Comment by necromanceranne** [comment]
The strongest use, and I feel the...probably only use of scatter laser is a charge to overwhelm someone with a point blank shot. I'd likened it to a baton rush. You're bulldozing an opponent fast enough they cannot respond. I think any other distance and you're really better off using a laser.

Ideally, that's not entirely off the table with this either. you're still rewarded for getting in close and overwhelming with brute force, because the damage will ramp up quickly. It will be just _slower_ in most cases. I don't think it will ultimately change their usage in either case. Just the potency, and maybe the impact of attrition will be felt more strongly on who you are fighting, assuming you don't win outright. Particularly if they're unlucky enough to hit sepsis.

**Comment by JupiterJaeden** [comment]
The revised version of this PR (with higher damage) is somewhat better than the original but I am still obligated to post this classic from 2021.

the fate of security

**Comment by Striders13** [comment]
you could make them craftable but not locked behind weapons tech (maybe just sec tech?). This still isn't good enough to be only unlocked 1 hour into the round imo.

**Comment by necromanceranne** [comment]
I'm not just reinventing improvised shells.

**Comment by CCC23** [comment]
litterally 1984 on our shotguns

**Comment by moocowswag** [comment]
Except batons are practically emp resistant since it only reduces their cell charge by a small flat amount (i dont think a full clip from an ion gun can drain it) and just gives them a temporary toggle on/off oofness

Meanwhile a single emp has the potential of permanently gimping a large amount of ammo you printed forcing you to go back and print more.

Idk i think batons are more reliable than point blank laser scatter especially for their higher damage per click and armor ignore.

Also laser guns do the same damage (and damage per hit is taken into account for rolling the dice on burn wounds so lasers do it more reliably) while also firing far faster.

A laser gun can fire multiple times per second (60+dps) while having better range meanwhile shotgun is fired once and needs to be activated to cycle to the next shell (a lot more micro intensive for less fire rate, damage per shot, and range and higher emp susceptibility and lower clip size)

You can literally down someone within the span of 2 seconds with a laser gun I dont think you can with nerfed laser scatters.

So i suggest one of the following

Make the damage falloff not start until after 2 tiles distance and add even more wound bonus so it becomes more of a de buff weapon for adding burn wounds to people sieging the brig (still not as good as laser gun but its something).

Or, since the ammo was not designed to be printed and replenished so easily, readd the crafting requirement and put its damage value somewhere higher to increase the risk/reward of getting empd or not.

**Comment by Cadunkus** [comment]
Buckshot was removed from security lathe making it harder to acquire.

Improvised shells were nerfed significantly.

Buckshot was nerfed on top of that.

Stop nerfing the shotgun. Seriously. It's actually a terrible weapon if you don't have technological shells thanks to these stupid PRs and now you want to get rid of the last reason to use it? No. Stop.

**Review by Jacquerel — CHANGES_REQUESTED** [review]

**Comment by Jacquerel** [comment]
You have fallen for the maintainers trap where I posted a review comment to add an early return but only highlighted the first two lines and not the entire proc

File: code/modules/projectiles/ammunition/ballistic/shotgun.dm [hunk 2]
```diff
@@ -121,14 +121,23 @@
...
 
+/obj/item/ammo_casing/shotgun/scatterlaser/emp_act(severity)
+	. = ..()
+	if(isnull(loaded_projectile) || !prob(40/severity))
+		return
+		name = "malfunctioning laser shell"
+		desc = "An advanced shotgun shell that uses a micro laser to replicate the effects of a scatter laser weapon in a ballistic package. The capacitor powering this assembly appears to be smoking."
+		projectile_type = /obj/projectile/beam/scatter/pathetic
+		loaded_projectile = new projectile_type(src)
+
```
Comment (Jacquerel) [inline code/modules/projectiles/ammunition/ballistic/shotgun.dm:140]:
```suggestion
	name = "malfunctioning laser shell"
	desc = "An advanced shotgun shell that uses a micro laser to replicate the effects of a scatter laser weapon in a ballistic package. The capacitor powering this assembly appears to be smoking."
	projectile_type = /obj/projectile/beam/scatter/pathetic
	loaded_projectile = new projectile_type(src)

```

Let me enable further web editing

Comment (necromanceranne) [reply code/modules/projectiles/ammunition/ballistic/shotgun.dm]: I'll save you the satisfaction and not do that.

**Comment by Robustitron** [comment]
I predicted this when improv was nerfed, dragons breath removal and beanbag slug stam damage halving are in the future, calling it now. 

10000 points and waiting til the end of the round, for something that has to be shot point blank to do equivalent damage of a single generic laser projectile, even if doubled, it's still not equivalent at point blank, to a generic laser's dps from afar.

The problem with the "shotgun for niche" idea is that no ones actually adding interesting niches, just nerfs. Even shotgun darts were nerfed so they don't stand out anymore.

Improvs are for breaking windows? Except worse than wall extinguishers considering crafting time? How exciting, I've literally never seen a player interested enough to craft one.

**Comment by necromanceranne** [comment]
okay

**Pull request merged**